packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -200,21 +200,41 @@ export const setSpanWithError = (
  * @param [extraOptions] additional options for the request
  */
 export const getRequestInfo = (
-  options: RequestOptions | string,
+  options: url.URL | RequestOptions | string,
   extraOptions?: RequestOptions
 ) => {
   let pathname = '/';
   let origin = '';
-  let optionsParsed: url.URL | url.UrlWithStringQuery | RequestOptions;
+  let optionsParsed: RequestOptions;
   if (typeof options === 'string') {
     optionsParsed = url.parse(options);
     pathname = (optionsParsed as url.UrlWithStringQuery).pathname || '/';
     origin = `${optionsParsed.protocol || 'http:'}//${optionsParsed.host}`;
     if (extraOptions !== undefined) {
       Object.assign(optionsParsed, extraOptions);
     }
+  } else if (options instanceof url.URL) {
+    optionsParsed = {
+      protocol: options.protocol,
+      hostname:
+        typeof options.hostname === 'string' && options.hostname.startsWith('[')
+          ? options.hostname.slice(1, -1)
+          : options.hostname,
+      path: `${options.pathname || ''}${options.search || ''}`,
+    };
+    if (optionsParsed.port !== '') {
```

Comment:
This is always true

Reply:
repl shows the following:
```
> new url.URL("http://foo.bar")
URL {
  href: 'http://foo.bar/',
  origin: 'http://foo.bar',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'foo.bar',
  hostname: 'foo.bar',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```
Seems URL uses empty string not `null` like `url.parse()`.

Reply:
`optionsParsed` is set on the lines above as a plain object, not a `URL`. Are you thinking of `options.port`?

Reply:
Thanks! the check has been changed.
I think we could improve variable naming a bit in here but I don't want to put too much into this PR.